boards/itsybitsy_m4/examples/sercom_interrupt.rs

```diff
@@ -0,0 +1,160 @@
+    mclk: &mut pac::MCLK,
+    uart_rx: impl Into<UartRx0>,
+    uart_tx: impl Into<UartTx0>,
+) -> Uart0 {
+    let gclk0 = clocks.gclk0();
+    let clock = &clocks.sercom0_core(&gclk0).unwrap();
+    let baud = baud.into();
+    let pads = uart::Pads::default().rx(uart_rx.into()).tx(uart_tx.into());
+    uart::Config::new(mclk, sercom0, pads, clock.freq())
+        .baud(baud, BaudMode::Fractional(Oversampling::Bits16))
+        .enable()
+}
+
+/// If you want to know more on ways to share data safely,
+/// give this a read: <https://docs.rust-embedded.org/book/concurrency/>
+static LED: Mutex<RefCell<Option<Pin<PA22, PushPullOutput>>>> = Mutex::new(RefCell::new(None));
+static SERIAL: Mutex<RefCell<Option<bsp::Uart>>> = Mutex::new(RefCell::new(None));
+
+#[entry]
+fn main() -> ! {
+    //basic setup
+    let mut dp = Peripherals::take().unwrap();
+    let mut core = CorePeripherals::take().unwrap();
+    let mut clocks = GenericClockController::with_internal_32kosc(
+        dp.GCLK,
+        &mut dp.MCLK,
+        &mut dp.OSC32KCTRL,
+        &mut dp.OSCCTRL,
+        &mut dp.NVMCTRL,
+    );
+
+    let pins = bsp::Pins::new(dp.PORT);
+    let mut delay = Delay::new(core.SYST, &mut clocks);
+
+    // custom sercom uart configuration
+    let mut serial_sercom0 = uart0(
+        &mut clocks,
+        Hertz(115200),
+        dp.SERCOM0,
+        &mut dp.MCLK,
+        pins.a5,
+        pins.a4,
+    );
+
+    // labeled "default" uart
+    let mut serial_sercom3 = bsp::uart(
+        &mut clocks,
+        Hertz(115200),
+        dp.SERCOM3,
+        &mut dp.MCLK,
+        pins.d0_rx,
+        pins.d1_tx,
+    );
+
+    // We must enable the interrupt flag by flag, in this case,
+    // we just want RXC
+    serial_sercom3.enable_interrupts(Flags::RXC);
+    let led = pins.d13.into_push_pull_output();
+    //move shared peripherals into the mutexes
+    cortex_m::interrupt::free(|cs| {
+        SERIAL.borrow(cs).replace(Some(serial_sercom3));
+        LED.borrow(cs).replace(Some(led));
+    });
+
+    unsafe {
+        // interrupt must be enabled and unmasked in the NVIC
+        // (Nested Vector Interrupt Controller) peripheral
+        core.NVIC.set_priority(interrupt::SERCOM3_2, 1);
+        NVIC::unmask(interrupt::SERCOM3_2);
+    }
+
+    loop {
+        // wait a second then send a byte
+        delay.delay_ms(1000u16);
+        let _ = nb::block!(serial_sercom0.write(b'A'));
+    }
+}
+
+#[interrupt]
+fn SERCOM3_2() {
+    // must read sercom data register or the interrupt
+    // won't trigger again
+    cortex_m::interrupt::free(|cs| {
+        if let Some(serial) = SERIAL.borrow(cs).borrow_mut().as_mut() {
+            if let Ok(c) = serial.read() {
+                let _ = nb::block!(serial.write(c));
+            }
+        }
+    });
+    toggle()
+}
+
+fn toggle() {
+    cortex_m::interrupt::free(|cs| {
+        if let Some(led) = LED.borrow(cs).borrow_mut().as_mut() {
+            led.toggle().unwrap();
```

Comment:
nit: other examples have been using the pattern `.ok()` since we know this toggle is infallible. I've seen it both ways though so up to you, just mentioning it
```suggestion
            led.toggle().ok();
```

Reply:
I think it looks cleaner with the ok(), so I think I will change that

Reply:
This hasn't been fixed yet